### PR TITLE
Add support for Chakra Host in sys.

### PR DESCRIPTION
**Pull request by paulvanbrenk**

### CODE REVIEW

File: src/compiler/sys.ts [hunk 1]
```diff
@@ -454,6 +469,53 @@ namespace ts {
                 }
             };
         }
+        function getChakraSystem(): System {
```
Comment (yuit) [inline src/compiler/sys.ts]: Add line

File: src/compiler/sys.ts [hunk 2]
```diff
@@ -454,6 +470,55 @@ namespace ts {
...
+                },
+                writeFile(path: string, data: string, writeByteOrderMark?: boolean) {
+                    // If a BOM is required, emit one
+                    if (writeByteOrderMark) {
+                        data = "\uFEFF" + data;
+                    }
+
+                    ChakraHost.writeFile(path, data);
+                },
+                resolvePath(path: string) {
```
Comment (RyanCavanaugh) [inline src/compiler/sys.ts]: Can these just be e.g. `resolvePath: ChakraHost.resolvePath` instead of wrappers?

File: src/compiler/sys.ts [hunk 3]
```diff
@@ -454,6 +470,55 @@ namespace ts {
...
+                    return ChakraHost.executingFile;
+                },
+                getCurrentDirectory() {
+                    return ChakraHost.currentDirectory;
+                },
+                readDirectory(path: string, extension?: string, exclude?: string[]) {
+                    return ChakraHost.readDirectory(path, extension, exclude);
+                },
+                exit(exitCode?: number) {
+                    ChakraHost.quit(exitCode);
```
Comment (RyanCavanaugh) [inline src/compiler/sys.ts]: `exitCode || 0` ?

**Comment by mhegazy** [comment]
:+1:

**Comment by DanielRosenwasser** [comment]
Should this be called `NativeHost`? It's not unreasonable to expect another host that exposes the same functionality.

**Comment by vladima** [comment]
:+1:

**Comment by yuit** [comment]
:+1: @DanielRosenwasser  is that what interface `System` for ?

File: src/compiler/sys.ts (hunk 2 above)
Comment (paulvanbrenk) [inline src/compiler/sys.ts]: Yes, that's a good idea.

**Pull request merged**